packages/js-drive/lib/identity/PublicKeyToIdentityIdStoreRepository.js

```diff
@@ -19,31 +21,56 @@ class PublicKeyToIdentityIdStoreRepository {
    * @return {Promise<PublicKeyToIdentityIdStoreRepository>}
    */
   async store(publicKeyHash, identityId, transaction = undefined) {
-    this.storage.put(
-      publicKeyHash,
-      identityId,
-      transaction,
-    );
+    const identityIdsSerialized = this.storage.get(publicKeyHash, transaction);
+
+    let identityIds = [];
+    if (identityIdsSerialized) {
+      identityIds = cbor.decode(identityIdsSerialized);
+    }
+
+    if (!identityIds.includes(identityId)) {
+      identityIds.push(identityId.toBuffer());
+
+      this.storage.put(
+        publicKeyHash,
+        cbor.encode(identityIds),
+        transaction,
+      );
+    }
 
     return this;
   }
 
   /**
-   * Fetch identity id by public key hash from database
+   * Fetch serialized identity ids by public key hash from database
    *
    * @param {Buffer} publicKeyHash
    * @param {MerkDbTransaction} [transaction]
    *
-   * @return {Promise<null|Identifier>}
+   * @return {Promise<Buffer|null>}
    */
   async fetch(publicKeyHash, transaction = undefined) {
```

Comment:
Let's keep `fetch` with decoding, like in other repos. And rename `fetchDeserialized` to `fetchBuffer`.